tests/ClientTest.php

```diff
@@ -534,43 +357,268 @@ public function testAddBreadcrumbStoresBreadcrumbReturnedByBeforeBreadcrumbCallb
         $this->assertSame([$breadcrumb2], $scope->getBreadcrumbs());
     }
 
-    public function testSetSerializer()
+    public function testHandlingExceptionThrowingAnException()
     {
-        $client = ClientBuilder::create()->getClient();
-        $serializer = $this->createMock(Serializer::class);
+        /** @var TransportInterface|\PHPUnit_Framework_MockObject_MockObject $transport */
+        $transport = $this->createMock(TransportInterface::class);
+
+        $client = new Client(new Options(), $transport, []);
+
+        Hub::getCurrent()->bindClient($client);
+
+        $transport->expects($this->once())
+            ->method('send')
+            ->with($this->callback(function ($event) {
+                /* @var Event $event*/
+                // Make sure the exception is of the careless exception and not the exception thrown inside
+                // the __set method of that exception caused by setting the event_id on the exception instance
+                $this->assertSame(CarelessException::class, $event->getException()[0]['type']);
 
-        $client->setSerializer($serializer);
+                return true;
+            }));
 
-        $this->assertSame($serializer, $client->getSerializer());
+        $client->captureException($this->createCarelessExceptionWithStacktrace(), Hub::getCurrent()->getScope());
     }
 
-    public function testSetReprSerializer()
+    /**
+     * @dataProvider convertExceptionDataProvider
+     */
+    public function testConvertException(\Exception $exception, array $clientConfig, array $expectedResult)
     {
-        $client = ClientBuilder::create()->getClient();
-        $serializer = $this->createMock(ReprSerializer::class);
+        $options = new Options($clientConfig);
 
-        $client->setRepresentationSerializer($serializer);
+        $assertHasStacktrace = $options->getAutoLogStacks();
 
-        $this->assertSame($serializer, $client->getRepresentationSerializer());
+        /** @var TransportInterface|\PHPUnit_Framework_MockObject_MockObject $transport */
+        $transport = $this->createMock(TransportInterface::class);
+
+        $transport->expects($this->once())
+            ->method('send')
+            ->with($this->callback(function (Event $event) use ($expectedResult, $assertHasStacktrace) {
+                $this->assertNotNull($event);
```

Comment:
This assert is unnecessary because the argument is typehinted to not accept `null` values